From 2be77bd99d5e5e7be41df80fb16f4ac836d875b5 Mon Sep 17 00:00:00 2001
From: Bahtiar `kalkin-` Gadimov <bahtiar@gadimov.de>
Date: Mon, 21 Aug 2017 13:07:33 +0200
Subject: [PATCH] Show domain name in titles

Signed-off-by: Bahtiar `kalkin-` Gadimov <bahtiar@gadimov.de>
---
 awesomerc.lua | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/awesomerc.lua b/awesomerc.lua
index 63431c72..d6da869e 100644
--- a/awesomerc.lua
+++ b/awesomerc.lua
@@ -584,6 +584,9 @@ client.connect_signal("request::titlebars", function(c)
     }
 end)

+-- Put the Qube name in front of all displayed names (tilebars, tasklists, ...)
+client.connect_signal("property::name", function(c) qubes.set_name(c) end)
+
 -- Enable sloppy focus, so that focus follows mouse.
 client.connect_signal("mouse::enter", function(c)
     c:emit_signal("request::activate", "mouse_enter", {raise = false})

